### DISCUSSION TITLE
Space for short-echo-tags.

### DISCUSSION THREAD

**Discussion by rquadling** [PHP-CS-Fixer/PHP-CS-Fixer#6821] [Q&A] [answered]
## Support question

Is there an option to enforce that `<?=$data?>` is `<?= $data ?>`, that is a space after the opening short echo tag (`<?=`) and a space before the closing short echo tag (`?>`)?

**Comment by SpacePossum** [comment]
Hi and thanks for your question!

Currently, AFAIK, we've no rules for this. Short echo open isn't the most popular here and we are not always that focused on supporting it. However, for the space after `<?=-` maybe https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/4435 can be extended (I haven't looked at the implementation details though). 

For the before I `?>`; I haven't looked yet.

**Comment by Wirone** [comment, marked as answer]
Please have in mind that Fixer is focused on modern PHP, that does not mix code with HTML markup (see [this comment](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/6788#issuecomment-1438688702)), that's why I'm going to close this discussion. Feel free to provide PR, but we can't assure it will be merged anyway.

**Reply by rquadling** [reply]
In honesty, that's completely fair. I was just looking for an cheat answer that meant I wouldn't need to write my own rule code! The whole "get my HTML template and PHP code all laid out correctly" is a nightmare. SOOO many things to deal with and cannot be done well. Biggest concern is how templates embed WAY too much logic, rather than splitting out dependencies and managing inclusion in a more constructive way. But that's IMHO. Other opinions exist and have as good / bad output as any.
